src/Stratis.SmartContracts.Core.Validation/Validators/Type/FieldDefinitionValidator.cs

```diff
@@ -0,0 +1,49 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Mono.Cecil;
+using Stratis.ModuleValidation.Net;
+
+namespace Stratis.SmartContracts.Core.Validation
+{
+    /// <summary>
+    /// Validates whether a type defines any fields
+    /// </summary>
+    public class FieldDefinitionValidator : ITypeDefinitionValidator
+    {
+         public IEnumerable<ValidationResult> Validate(TypeDefinition type)
+         {
+             // Don't apply to nested enums or structs
+             if (type.IsNested && type.IsValueType)
+                 return Enumerable.Empty<TypeDefinitionValidationResult>();
+
+            var errors = new List<TypeDefinitionValidationResult>();
+
+             foreach (FieldDefinition field in type.Fields)
```

Comment:
The `if` below can be combined in the `foreach`?

Reply:
The logic here is important so I want to keep it readable. IMO this is easier when it's within the body of the loop alone, rather than in a `Where` clause + in the body of the loop.